src/QueryBuilder.js

```diff
@@ -1566,6 +1570,229 @@ QueryBuilder.prototype.returning = queryMethod('returning');
+  } else if (_.isObject(jsonObjectOrFieldExpression)) {
+    var refValQuery = ["(", fieldReference, ")::jsonb ", operator, " ?::jsonb"].join("");
+    return this.whereRaw(refValQuery, JSON.stringify(jsonObjectOrFieldExpression));
+  }
+  throw new Error("Invalid right hand expression.");
+};
+
+
+/**
+ * Json equality comparison.
+ *
+ * Also supports having field expression in both sides of equality.
+ *
+ * ```js
+ * Person
+ *   .query()
+ *   .whereJsonEquals('additionalData.myDogs', 'additionalData.dogsAtHome')
+ *   .then(function (person) {
+ *     // oh joy! these persons has all their dogs at home!
+ *   });
+ *
+ * Person
+ *   .query()
+ *   .whereJsonEquals('additionalData.myDogs[0]', { name: "peter"})
+ *   .then(function (person) {
+ *     // these persons' first dog name is "peter" and the dog has no other
+ *     // attributes, but its name
+ *   });
+ * ```
+ *
+ * @param fieldExpression {String} Reference to column / jsonField.
+ * @param jsonObjectOrFieldExpression {Object|Array|String} Reference to column / jsonField or json object.
+ * @returns {QueryBuilder}
+ */
+QueryBuilder.prototype.whereJsonEquals = function (fieldExpression, jsonObjectOrFieldExpression) {
+  return this.whereJsonObject(fieldExpression, "=", jsonObjectOrFieldExpression);
+};
+
+/**
+ * Json filters all results where left hand operator is superset of the right hand operand.
+ *
+ * ```js
+ * Person
+ *   .query()
+ *   .whereJsonSupersetOf('additionalData.myDogs', 'additionalData.dogsAtHome')
+ *   .then(function (person) {
+ *     // oh joy! these persons has all their dogs at home!
+ *     // But there might be actually some extra dogs there since
+ *     // we requested with subset
+ *   });
+ *
+ * Person
+ *   .query()
+ *   .whereJsonSupersetOf('additionalData.myDogs[0]', { name: "peter"})
+ *   .then(function (person) {
+ *     // these persons' first dog name is "peter", but the dog might have
+ *     // additional attributes as well
+ *   });
+ * ```
+ *
+ * Object and array are always their own supersets.
+ *
+ * For arrays this mean that all arrays of left side matches if it has all the elements
+ * listed in the right hand side. e.g.
+ *
+ * [1,2,3] isSuperSetOf [2] => true
+ * [1,2,3] isSuperSetOf [2,1,3] => true
+ * [1,2,3] isSuperSetOf [2,null] => false
+ * [1,2,3] isSuperSetOf [] => true
+ *
+ * @param fieldExpression {String} Reference to column / jsonField, which is tested being supoerset.
+ * @param jsonObjectOrFieldExpression {Object|Array|String} to which to compare.
+ * @returns {QueryBuilder}
+ */
+QueryBuilder.prototype.whereJsonSupersetOf = function (fieldExpression, jsonObjectOrFieldExpression) {
+  return this.whereJsonObject(fieldExpression, "@>", jsonObjectOrFieldExpression);
+};
+
+/**
+ * Json filters all results where left hand operator is subset of the right hand operand.
+ *
+ * Object and array are always their own subsets.
+ *
+ * see {QueryBuilder.prototype.whereJsonSupersetOf}
```

Comment:
Did you test that this becomes a link by running `grunt jsdoc`? If this doesn't work you can use `{@link QueryBuilder.prototype.whereJsonSupersetOf}